libs/blocks/video-metadata/video-metadata.js

```diff
@@ -0,0 +1,116 @@
+      break;
+    default:
+      window.lana.log(`VideoMetadata -- Unknown BroadcastEvent property: ${blockKey}`);
+      break;
+  }
+}
+
+function addClipField(videoObj, blockKey, blockValue) {
+  const [, num, key] = blockKey.match(CLIP_RE);
+  const i = num - 1;
+  if (!videoObj.hasPart) videoObj.hasPart = [];
+  if (!videoObj.hasPart[i]) videoObj.hasPart[i] = { '@type': 'Clip' };
+  switch (key) {
+    case 'start-offset':
+    case 'end-offset':
+      videoObj.hasPart[i][camelize(key)] = parseInt(blockValue, 10);
+      break;
+    case 'name':
+    case 'url':
+      videoObj.hasPart[i][camelize(key)] = blockValue;
+      break;
+    default:
+      window.lana.log(`VideoMetadata -- Unhandled Clip property: ${blockKey}`);
+      break;
+  }
+}
+
+function addSeekToActionField(videoObj, blockKey, blockValue) {
+  const [, key] = blockKey.match(SEEK_TO_ACTION_RE);
+  if (!videoObj.potentialAction) videoObj.potentialAction = { '@type': 'SeekToAction' };
+  switch (key) {
+    case 'target':
+      videoObj.potentialAction.target = blockValue;
+      break;
+    case 'start-offset-input':
+      videoObj.potentialAction['startOffset-input'] = blockValue;
+      break;
+    default:
+      window.lana.log(`VideoMetadata -- Unhandled SeekToAction property: ${blockKey}`);
+      break;
+  }
+}
+
+export function createVideoObject(blockMap) {
+  const video = {
+    '@context': 'https://schema.org',
+    '@type': 'VideoObject',
+  };
+  Object.entries(blockMap).forEach(([k, v]) => {
+    const blockVal = v.content && v.content.textContent.trim();
+    if (!blockVal) return;
+    const blockKey = k && k.replaceAll(' ', '-');
+    switch (true) {
+      case blockKey === 'content-url':
+      case blockKey === 'description':
+      case blockKey === 'duration':
+      case blockKey === 'embed-url':
+      case blockKey === 'expires':
+      case blockKey === 'name':
+      case blockKey === 'regions-allowed':
+      case blockKey === 'upload-date':
+        video[camelize(blockKey)] = blockVal;
+        break;
+      case blockKey === 'thumbnail-url':
+        video.thumbnailUrl = blockVal.split(LINES2ARRAY_SPLIT_RE);
+        if (video.thumbnailUrl.length < 2) video.thumbnailUrl = blockVal;
+        break;
+      case BROADCAST_EVENT_RE.test(blockKey):
+        addBroadcastEventField(video, blockKey, blockVal);
+        break;
+      case CLIP_RE.test(blockKey):
+        addClipField(video, blockKey, blockVal);
+        break;
+      case SEEK_TO_ACTION_RE.test(blockKey):
+        addSeekToActionField(video, blockKey, blockVal);
+        break;
+      default:
+        window.lana.log(`VideoMetadata -- Unhandled VideoObject property: ${blockKey}`);
+        break;
+    }
+  });
+  return video;
+}
+
+export default function init(el) {
+  const kv = getMetadata(el);
+  const obj = createVideoObject(kv);
+  const script = createTag('script', { type: 'application/ld+json' }, JSON.stringify(obj));
+  document.head.append(script);
+  el.remove();
+}
```

Comment:
```suggestion
export default function init(el) {
  const kv = getMetadata(el);
  el.remove();
  const obj = createVideoObject(kv);
  if (!obj) return;
  const script = createTag('script', { type: 'application/ld+json' }, JSON.stringify(obj));
  document.head.append(script);
}
```

Small nits.

Reply:
I moved element deletion as suggested. [97f6de9](https://github.com/adobecom/milo/pull/611/commits/97f6de9eca5a760bf0c0392ae0558bdeb96840d6)

I did not add early return on falsey obj.
It results in an unreachable branch that I can't test/cover.
createVideoObject always returns video object with at least content/type.
This was done on purpose as authors should see validation errors (e.g. schema.org) if they wrote a bad block.

Reply:
> It results in an unreachable branch that I can't test/cover.

You should be able to cover that, no?

parahphrasing...
```js
const videoJson = init(myBadEl);
expect(videoJson).to.be.undefined;
```

> This was done on purpose as authors should see validation errors (e.g. schema.org) if they wrote a bad block.

I understand the thought, but the reality is: 8 out of 10 times, authors won't check their schema and it will result in live errors. We (and the SEO team) would rather have it not exist at all than have an incorrect schema in the wild.

Reply:
I can rewrite so that there is no object returned if there are no valid key/values at all.

This doesn't change fact that this block and prior metadata-related PRs have little to no value validation.
That would increase scope (and code) significantly.
Authors *should* check.
IMO validation should be part of sidekick.

Reply:
Resolved at d6c9a7c